### ISSUE TITLE
Unable to get location while offline, and using last location will cause crash

### ISSUE THREAD

**Issue by sjl872964789** [forrestguice/SuntimesWidget#420]
**Describe the bug**
If the first time you open the suntimeswidget is offline, the get location function cannot be used and keeps the user waiting, and if you click use previous location after waiting for a long time, it will crash.

**To Reproduce**
1. Turn off network
2. Install SuntimesWidget and open app
2. Click "location"
3. Click "update"
4. A long wait
5. Pop up the prompt box of "unable to get a fix"
6. Click on "USE LAST LOCATION"
7.Crash

**Expected behavior**
The user will be prompted to open the network without crashing

**Smartphone (please complete the following information):**
 - Device: HUAWEI nova 5 pro
 - OS: EMUI 9.1.1
 - Rooted: No
 - Version: v0.12.10

**Exception**
```
java.lang.NullPointerException
	at com.forrestguice.suntimeswidget.calculator.core.Location.<init>(Location.java:100)
	at com.forrestguice.suntimeswidget.SuntimesActivity$8.updateUI(SuntimesActivity.java:827)
	at com.forrestguice.suntimeswidget.getfix.GetFixHelper.fallbackToLastLocation(GetFixHelper.java:167)
	at com.forrestguice.suntimeswidget.getfix.GetFixHelper$KeepTryingDialog$1.onClick(GetFixHelper.java:402)
	at com.android.internal.app.AlertController$ButtonHandler.handleMessage(AlertController.java:166)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:136)
	at android.app.ActivityThread.main(ActivityThread.java:5135)
	at java.lang.reflect.Method.invokeNative(Native Method)
	at java.lang.reflect.Method.invoke(Method.java:515)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:798)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:614)
	at dalvik.system.NativeStart.main(Native Method)
```

**Additional context**
[record.zip](https://github.com/forrestguice/SuntimesWidget/files/4744282/record.zip)

**Comment by forrestguice** [comment]
Thanks for reporting this crash. It will be fixed in v0.12.11.

A few notes..

Suntimes doesn't use network permissions for anything, so it isn't necessary for you to be "online" to get location. The app will use whatever location provider is available. If the network provider is disabled that means "GPS only", which will eventually time out without a good signal. It sounds like this is a necessary condition for triggering the bug.

It seems the "last known location" is null for some reason, and we're not checking for that. I guess this indicates that none of the location providers have been used at any point. I'll try reproducing this with a fresh emulator.

**Comment by sjl872964789** [comment]
Thank you for your reply, I think your analysis is correct, when I open the network and switch the location mode to "device only", I will also fall into the situation of infinite load. If you have any other progress on this issue, please let me know at any time, thank you again!
